homeassistant/components/input_text/__init__.py

```diff
@@ -235,7 +236,7 @@ def state(self):
         """Return the state of the component."""
         return self._current_value
 
-    @property
+    @cached_property
```

Comment:
This cache needs to be cleared when `async_update_config` is called, as it can update the values.

Reply:
I don't think we have a way to do that so let's not cache this one

Reply:
I think I need more coffee this morning

Reply:
reverted the correct thing now, pushed to the correct branch, and should be ok if mypy passes